### ISSUE TITLE
Confusing error message - consider making `mut actual` mutable: `mut mut actual`

### ISSUE THREAD

**Issue by cyplo** [rust-lang/rust#48933]
Heya ! 
I'm playing with different ways of passing initialization code and found a misleading compiler error:
https://play.rust-lang.org/?gist=3a24c7062bb9b48bcdd5ceea6dc6174e&version=stable
```
29 |     let mut actual: String = "".to_string();
   |         ---------- help: consider making `mut actual` mutable: `mut mut actual`
30 |     let calcrustlator = Calcrustlator::with_expression(input, &move |expression| {
31 |         actual = String::from(expression);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
It also includes the following advice:
```
   |
29 |     let mut actual: String = "".to_string();
   |         ---^^^^^^^
   |         |
   |         help: remove this `mut`
   |
```
which seems to be the exact opposite of the first one

When someone tries to actually implement the suggested fix the error then becomes:
```
error: expected identifier, found keyword `mut`
  --> src/main.rs:29:13
   |
29 |     let mut mut actual: String = "".to_string();
   |             ^^^
```

Thank you for reading and have a great day !

**Comment by rkruppe** [comment]
Thanks for the report! Luckily it appears this has [already been fixed in nightly](https://play.rust-lang.org/?gist=2517bfc5f0f6c574a7e60e3209040bd3&version=nightly).

**Comment by cyplo** [comment]
Cool, thank you :)
